web_monitoring/db.py

```diff
@@ -1,47 +1,608 @@
+    -------
+    response : dict
+    """
+    url = (f'{api_url()}/pages/{page_id}/changes/'
+           f'{from_version_id}..{to_version_id}/annotations')
+    res = requests.get(url, auth=auth())
+    res.raise_for_status()
+    result = res.json()
+    # In place, replace datetime strings with datetime objects.
+    for a in result['data']:
+        a['created_at'] = parse_timestamp(a['created_at'])
+        a['updated_at'] = parse_timestamp(a['updated_at'])
+    return result
+
+
+@ensure_credentials
+def post_annotation(annotation, page_id, to_version_id, from_version_id=''):
+    """
+    Submit updated annotations for a change between versions.
+
+    Parameters
+    ----------
+    annotation : dict
+    page_id : string
+    to_version_id : string
+    from_version_id : string, optional
+        If from_version_id is not given, it will be treated as version
+        immediately prior to ``to_version``.
+
+    Returns
+    -------
+    response : dict
     """
-    url = WEB_MONITORING_SHOW_IMPORT_API.format(db_url=settings['db_url'],
-                                                import_id=import_id)
-    return requests.get(url,
-                        auth=(settings['db_email'], settings['db_password']))
+    url = (f'{api_url()}/pages/{page_id}/changes/'
+           f'{from_version_id}..{to_version_id}/annotations')
+    res = requests.post(url, auth=auth(),
+                        headers={'Content-Type': 'application/json'},
+                        data=json.dumps(annotation))
+    res.raise_for_status()
+    return res.json()
+
+
+@ensure_credentials
+def get_annotation(annotation_id, page_id, to_version_id, from_version_id=''):
+    """
+    Get a specific Annontation.
+
+    Parameters
+    ----------
+    annotation_id : string
+    page_id : string
+    to_version_id : string
+    from_version_id : string, optional
+        If from_version_id is not given, it will be treated as version
+        immediately prior to ``to_version``.
+
+    Returns
+    -------
+    response : dict
+    """
+    url = (f'{api_url()}/pages/{page_id}/changes/'
+           f'{from_version_id}..{to_version_id}/annotations/{annotation_id}')
+    res = requests.get(url, auth=auth())
+    res.raise_for_status()
+    result = res.json()
+    # In place, replace datetime strings with datetime objects.
+    data = result['data']
+    data['created_at'] = parse_timestamp(data['created_at'])
+    data['updated_at'] = parse_timestamp(data['updated_at'])
+    return result
+
+
+### CONVENIENCE FUNCTIONS ###
+
+
+@ensure_credentials
+def fetch_version_content(version_id):
+    """
+    Download the saved content from a given Version.
+
+    Parameters
+    ----------
+    version_id : string
+
+    Returns
+    -------
+    content : bytes
+    """
+    url = f'{api_url()}/versions/{version_id}'
+    res = requests.get(url, auth=auth())
+    assert res.ok
+    content_uri = res.json()['data']['uri']
+    return requests.get(content_uri).content
```

Comment:
Would it make sense to return `text` or `content` based on the `Content-Type` header?